twitter-clone/src/components/AuthForm.tsx

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+interface FormProps {
+  onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
+  headerTitle: string;
+  footer: React.ReactNode;
+  children: React.ReactNode;
+}
+
+const AuthForm: React.FC<FormProps> = ({
+  onSubmit,
+  children,
+  headerTitle,
+  footer,
+}) => (
+  <div className="min-h-screen flex items-center justify-center bg-gray-50 py-12 px-4 sm:px-6 lg:px-8">
+    <div className="container mx-auto max-w-md border-2 border-gray-300 bg-gray-100 rounded-md p-4 mt-4">
+      <h1 className="text-center text-2xl font-bold mb-4">{headerTitle}</h1>
+      <form onSubmit={onSubmit}>{children}</form>
+      <div className="text-center mt-4">{footer}</div>
+    </div>
+  </div>
```

Comment:
it's not really best approach for re-usable component.
We reuse only 6 lines, but instead we have 2 similar pages SignupPage and LoginPage, there a lot of similar fields and similar logic.
In this way it's better to create reuseable form with fields component, and hide/show needed inputs via props.
Or other way it's move inputs to separate components and re-use them no both pages.

Reply:
Left as is. Wrapper was created to avoid styles duplication.